can-stache-bindings.js

```diff
@@ -351,27 +355,53 @@ var behaviors = {
 		// and can-xxx (anything starting with can-), this callback will be run.  Inside, its setting up an event handler
 		// that calls a method identified by the value of this attribute.
 		event: function(el, data) {
+
 			// Get the `event` name and if we are listening to the element or viewModel.
 			// The attribute name is the name of the event.
-			var attributeName = data.attributeName,
+			var attributeName = encoder.decode( data.attributeName ),
+				// the name of the event we are binding
+				event,
+				// if we are binding on the element or the VM
+				bindingContext;
+
+			// legacy binding
+			if(attributeName.indexOf('can-') === 0) {
+				event = attributeName.substr("can-".length);
+				bindingContext = el;
+			} else if(attributeName.indexOf("on:") === 0 ) {
```

Comment:
I think there wouldn't be much benefit, but we might want to cache `":on"` in the outer scope so we can do:

`.indexOf(onMatchStr)` and `onMatchStr.length`